src/bpmailsend.c

```diff
@@ -51,30 +53,57 @@ static int bpmailsend(void) {
      */
     if (content_size > UINT_MAX) {
         (void)fprintf(stderr, "content too large to send\n");
+        free(content);
+        return EXIT_FAILURE;
+    }
+
+    // Compress content using zlib
+    const char magic[] = "ZLIB";  // magic header to identify compressed content
+    uLongf compressed_size = compressBound((uLongf)content_size);
+    Bytef *compressed = malloc(strlen(magic) + compressed_size);
+
+    if (!compressed) {
```

Comment:
```suggestion
    if (compressed == NULL) {
```
Our project's preference